rules/internal/utils/extension.go

```diff
@@ -200,3 +203,33 @@ func SplitResourceTypeName(typ string) (service string, typeName string, ok bool
 
 	return
 }
+
+// FindResourceChildren attempts to search for other resources defined in the
+// package that are parented by the given resource.
+func FindResourceChildren(parent *apb.ResourceDescriptor, file *desc.FileDescriptor) []*apb.ResourceDescriptor {
+	pats := parent.GetPattern()
+	if len(pats) == 0 {
+		return nil
+	}
+	// Use the first pattern in the resource because:
+	// 1. Patterns cannot be rearranged, so this is the true first pattern
+	// 2. The true first pattern is the one most likely to be used as a parent.
+	first := pats[0]
+
+	var children []*apb.ResourceDescriptor
+	files := append(file.GetDependencies(), file)
```

Comment:
> it would actually be dependents that would import the parent, correct?

Not necessarily. `a.proto` containing a child resource does not need to import the `b.proto` containing parent resource definition to actually define the child resource via `google.api.resource`. With this in mind, starting the search from the child resource's file (if split) does not mean we will find the parent resources. See below for the subtleties of this utility API that I think I need to improve.

I'll assert that the most common proto layouts are as follows:
* `service.proto` contains all request messages and all resources
* `service.proto` contains all request messages, some resources, imports files w/other resources
* `service.proto` contains all request messages, no resources, imports files w/all resources

Super edge case that i've seen once or twice:
* `service.proto` contains no request message, no resources, imports files w/all resources/request messages -> `requests.proto` defines/imports all resources

If we always use the `service.proto` or `requests.proto` file as `file` here, we will always find all resources.

Focusing on the rule implemented here specifically, we are always coming from linting the Standard Delete request message, which is defined in the `service.proto` (or sometimes a `requests.proto`) and the file we are passing in this API is the one of those. With the above assertions true, I'd say that for a majority of cases the algorithm works.

All of That said, **this isn't an intuitive API if that nuance is necessary**.

WDYT?

Reply:
there's also a pattern of "service.proto" and "resources.proto". But those cases would still be caught in your check.

sounds good to me. Maybe the solution to all of this is to have a single proto for the linter.

Reply:
The one case we cannot cover is resources within the same package split into separate service files, where one is a child of the other, but never depends on the parent's resource message concretely. This is a linter shortcoming in general, there is no way to see all of the files in the package, just what the specific element can actually see.

At least that is the case currently, we could potentially improve the linter to make all files it receives accessible to all rules. 

Anyways, this is as good as it will get for now.

Thanks for the discussion!